prime-router/src/main/kotlin/tokens/TokenAuthentication.kt

```
@@ -71,13 +70,7 @@ class TokenAuthentication(val jtiCache: JtiCache) : Logging {
                 return false
             }
             return jtiCache.isJTIOk(jti, expiresAt) // check for replay attacks
-        } catch (ex: JwtException) {
-            logErr(actionHistory, "Rejecting SenderToken JWT: $ex")
-            return false
-        } catch (e: IllegalArgumentException) {
-            logErr(actionHistory, "Rejecting SenderToken JWT: $e")
-            return false
-        } catch (e: NullPointerException) {
+        } catch (e: Exception) {
```

Allow it to truly fail if it's an exception we don't recognize (it is then a true exception to expected behavior). The JWT and Illegal argument exceptions are expected bad states that have meaning (returning false). All others are something else that should not be swallowed.

---

I'm good with taking out the catch for the `NullPointerException`, especially if we understand where it's coming from and handle it in a more correct way (e.g. `if null`)

---

But do we *really* want to throw an exception here? If we've given a bad JWT isn't the best response to return a 401 and be done with it? Because if we throw exceptions we'll wind up alerting people...

---

I'm still kinda meh on that but don't feel strongly enough about it, so will update with your initial suggestion. Thanks for the feedback and discussion!

---

NOPE -- that doesn't work either, apparently the Jwts class raises an NPE rather than a JwtException. IMO that's broken, but it's not our library so :shrug:

Will close this PR and mark the item on the code scanning report as a workaround for a library.

---

ok.. and yeah that is not great that the lib raises a null pointer exception.....